collector/textfile_test.go

```diff
@@ -0,0 +1,126 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package collector
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"github.com/prometheus/common/log"
+	kingpin "gopkg.in/alecthomas/kingpin.v2"
+)
+
+type collectorAdapter struct {
+	Collector
+}
+
+// Describe implements the prometheus.Collector interface.
+func (a collectorAdapter) Describe(ch chan<- *prometheus.Desc) {
+	// We have to send *some* metric in Describe, but we don't know which ones
+	// we're going to get, so just send a dummy metric.
+	ch <- prometheus.NewDesc("dummy_metric", "Dummy metric.", nil, nil)
+}
+
+// Collect implements the prometheus.Collector interface.
+func (a collectorAdapter) Collect(ch chan<- prometheus.Metric) {
+	err := a.Update(ch)
+	if err != nil {
+		panic(fmt.Sprintf("failed to update collector: %v", err))
+	}
+}
+
+func TestTextfileCollector(t *testing.T) {
+	tests := []struct {
+		path string
+		out  string
+	}{
+		{
+			path: "fixtures/textfile/no_metric_files",
+			out:  "fixtures/textfile/no_metric_files.out",
+		},
+		{
+			path: "fixtures/textfile/two_metric_files",
+			out:  "fixtures/textfile/two_metric_files.out",
+		},
+		{
+			path: "fixtures/textfile/nonexistent_path",
+			out:  "fixtures/textfile/nonexistent_path.out",
+		},
+		{
+			path: "fixtures/textfile/client_side_timestamp",
+			out:  "fixtures/textfile/client_side_timestamp.out",
+		},
+		{
+			path: "fixtures/textfile/different_metric_types",
+			out:  "fixtures/textfile/different_metric_types.out",
+		},
+		{
+			path: "fixtures/textfile/inconsistent_metrics",
+			out:  "fixtures/textfile/inconsistent_metrics.out",
+		},
+		{
+			path: "fixtures/textfile/histogram",
+			out:  "fixtures/textfile/histogram.out",
+		},
+		{
+			path: "fixtures/textfile/histogram_extra_dimension",
+			out:  "fixtures/textfile/histogram_extra_dimension.out",
+		},
+		{
+			path: "fixtures/textfile/summary",
+			out:  "fixtures/textfile/summary.out",
+		},
+		{
+			path: "fixtures/textfile/summary_extra_dimension",
+			out:  "fixtures/textfile/summary_extra_dimension.out",
+		},
+	}
+
+	for i, test := range tests {
+		mtime := 1.0
+		c := &textFileCollector{
+			path:  test.path,
+			mtime: &mtime,
+		}
+
+		// Suppress a log message about `nonexistent_path` not existing, this is
+		// expected and clutters the test output.
+		log.AddFlags(kingpin.CommandLine)
```

Comment:
We haven't switched the wmi_exporter to kingpin for flags yet, so this probably doesn't make sense